src/WebJobs.Script/Binding/Http/HttpBinding.cs

```diff
@@ -39,7 +40,11 @@ public override Collection<CustomAttributeBuilder> GetCustomAttributes(Type para
         public override Task BindAsync(BindingContext context)
         {
             HttpRequest request = (HttpRequest)context.TriggerValue;
-
+            if (context.Value is HttpResponseMessage invocationResponseMessage)
```

Comment:
Rather than duplicate this line (also set below), why not just put the below http response creation code in an if block and only do it if the value isn't already an http response? That way no duplicated code.

Reply:
updated. Let me know what you think.